transfers/util.py

```
@@ -53,35 +54,190 @@
 }
 
 
+class MeasuringPointEstimator:
+    def __init__(self):
+        df = read_csv("WaterLevels")
+        df["DateMeasured"] = pd.to_datetime(df["DateMeasured"], errors="coerce")
+        self._df = df.dropna(subset=["DateMeasured"])
+
+    def estimate_measuring_point_height(
+        self, row
+    ) -> tuple[float, str, datetime | None]:
+        mph = row.MPHeight
+        mph_desc = row.MeasuringPoint
+        df = self._df[self._df["PointID"] == row.PointID]
+        df = df.sort_values("DateMeasured")
+        if mph is None:
+            logger.info(
+                f"No MPHeight found for PointID: {row.PointID}. Estimating from measurements."
+            )
+            mphs = []
+            start_dates = []
+            mph_descs = []
+
+            if len(df) == 0:
+                logger.warning(f"No measurements found for PointID: {row.PointID}.")
+            else:
+                # try to estimate mpheight from measurements
+                for m in df.itertuples():
+                    mphi = m.DepthToWater - m.DepthToWaterBGS
```

This will run into an error if the row doesn't have either `DepthToWater` or `DepthToWaterBGS`. Can you put an if statement here? And if neither are found (and there's no mp height) should you raise an error?

---

I have not experienced that issue. This is technical debt that may or may not need to be addressed